diff --git a/scripts/tf_cnn_benchmarks/preprocessing.py b/scripts/tf_cnn_benchmarks/preprocessing.py
index e2a50db..e080f56 100644
--- a/scripts/tf_cnn_benchmarks/preprocessing.py
+++ b/scripts/tf_cnn_benchmarks/preprocessing.py
@@ -503,10 +503,10 @@ class RecordInputImagePreprocessor(object):
         if not file_names:
           raise ValueError('Found no files in --data_dir matching: {}'
                            .format(glob_pattern))
-        ds = tf.data.TFRecordDataset.list_files(file_names)
+        ds = tf.data.TFRecordDataset.list_files(glob_pattern)
         ds = ds.apply(
             interleave_ops.parallel_interleave(
-                tf.data.TFRecordDataset, cycle_length=10))
+                tf.data.TFRecordDataset, cycle_length=1, prefetch_input_elements=10))
         if cache_data:
           ds = ds.take(1).cache().repeat()
         counter = tf.data.Dataset.range(self.batch_size)
